apps/qubit/modules/settings/actions/ldapAction.class.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+/*
+ * This file is part of the Access to Memory (AtoM) software.
+ *
+ * Access to Memory (AtoM) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Access to Memory (AtoM) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Access to Memory (AtoM).  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+class SettingsLdapAction extends DefaultEditAction
+{
+  // Arrays not allowed in class constants
+  public static
+    $NAMES = array(
+      'ldapHost',
+      'ldapPort',
+      'ldapBaseDn');
+
+  protected function earlyExecute()
+  {
+
+  }
+
+  protected function addField($name)
+  {
+    switch ($name)
+    {
+      case 'ldapHost':
+      case 'ldapPort':
+      case 'ldapBaseDn':
+        $default = ($name == 'ldapPort') ? 389 : '';
+        if (null !== $this->{$name} = QubitSetting::getByName($name))
+        {
+          $default = $this->{$name}->getValue(array('sourceCulture' => true));
+        }
+        $this->form->setDefault($name, $default);
+
+        $validator = ($name == 'ldapPort') ? new sfValidatorInteger(array('min' => 1, 'max' => 65535)) : new sfValidatorPass;
+        $this->form->setValidator($name, $validator);
+        $this->form->setWidget($name, new sfWidgetFormInput);
+
+        break;
+    }
+  }
+
+  protected function processField($field)
+  {
+    switch ($name = $field->getName())
+    {
+      case 'ldapHost':
+      case 'ldapPort':
+      case 'ldapBaseDn':
+        if (null === $this->{$name})
+        {
+          $this->{$name} = new QubitSetting;
+          $this->{$name}->name = $name;
+          $this->scope = 'ldap';
+        }
+        $this->{$name}->setValue($field->getValue(), array('culture' => 'en'));
```

Comment:
The array should pass `'sourceCulture' => true` to work with installations where `default_culture != 'en'` I believe.